message_ix/util/__init__.py

```diff
@@ -232,3 +233,23 @@ def expand_dims(scenario: Scenario, name, **data):
 
         # Add the expanded data
         scenario.add_par(name, new_data)
+
+
+def sankey_mapper(df, year, region, flows_not_needed=[], variables_not_needed=[]):
+    df_filtered = df.filter(region=region + "*", year=year)
+
+    mapping = {}
+    for var in df_filtered["variable"]:
+        if gvc(var, 0) == "in":
+            mapping[var] = (gvc(var, [1, 2], join=True), gvc(var, [3, 4], join=True))
+        if gvc(var, 0) == "out":
+            mapping[var] = (gvc(var, [3, 4], join=True), gvc(var, [1, 2], join=True))
+
+    for k in mapping.keys():
```

Comment:
If above works, then this full block can be removed

Reply:
Not quite sure my edits support the same range of exclusion as the original did, but I'm not quite sure what the difference between "flow" and "variable" is and you can now exclude things like "final|electricity", which seems to be what we want?

Reply:
I think it makes fully sense to merge both as you did. I used an older piece of code for that. And the reason for distinguishing "flow" and "variable" was, if I remember correctly, that some reports could have something like emissions or costs included. But even in this case one "exclude" list would have been sufficient ;-)

Reply:
Thanks for confirming :)